platform/darwin/src/NSExpression+MGLAdditions.mm

```diff
@@ -878,9 +878,25 @@ + (instancetype)expressionWithMGLJSONObject:(id)object {
                 NSExpression *expression = [NSExpression expressionWithMGLJSONObject:argumentObjects[index]];
                 NSMutableDictionary *attrs = [NSMutableDictionary dictionary];
                 if ((index + 1) < argumentObjects.count) {
-                    attrs = argumentObjects[index + 1];
+                    attrs = [NSMutableDictionary dictionaryWithDictionary:argumentObjects[index + 1]];
                 }
                 
+                if (attrs.count) {
+                    if (attrs[MGLFontNamesAttribute]) {
+                        NSArray *fontNames = (NSArray *)attrs[MGLFontNamesAttribute];
+                        attrs[MGLFontNamesAttribute] = fontNames[1];
+                    }
+                    if (attrs[MGLFontColorAttribute] && [attrs[MGLFontColorAttribute] isKindOfClass:[NSArray class]]) {
+                        NSArray *colorArray = attrs[MGLFontColorAttribute];
+                        if ([colorArray[0] isEqualToString:@"rgb"] || [colorArray[0] isEqualToString:@"rgba"]) {
+                            NSArray *colorArguments = [colorArray subarrayWithRange:NSMakeRange(1, colorArray.count - 1)];
+                            NSArray *subexpressions = MGLSubexpressionsWithJSONObjects(colorArguments);
+                            UIColor *color = [NSExpression mgl_colorWithRGBComponents:subexpressions];
```

Comment:
Above, we allow the array’s first element to be `rgba` above, but here we call `+[NSExpression mgl_colorWithRGBComponents:]` instead of `+[NSExpression mgl_colorWithRGBAComponents:]`, so the alpha gets replaced with 1.0.

Reply:
`mgl_colorWithRGBComponents` accounts for alpha values. Since I don't want an alpha expression using this function is straightforward. Creates the color, if it has an alpha uses that if not then adds 1.0.

Reply:
`+[NSExpression mgl_colorWithRGBComponents:]` doesn’t account for the specified alpha. It hard-codes an alpha of 1.0. You’re thinking of ` +[NSExpression mgl_colorWithRGBAComponents:]`.

Reply:
I think this code accounts for the alpha if an alpha is passed:
https://github.com/mapbox/mapbox-gl-native/blob/10e1e63aee95f579f951e3c2001640a268d3d95c/platform/ios/src/UIColor%2BMGLAdditions.mm#L50-L70

Reply:
Oh interesting, that doesn’t appear to be the case on macOS:

https://github.com/mapbox/mapbox-gl-native/blob/10e1e63aee95f579f951e3c2001640a268d3d95c/platform/macos/src/NSColor%2BMGLAdditions.mm#L45